src/vos/vos_internal.h

```diff
@@ -60,6 +60,8 @@ extern struct dss_module_key vos_module_key;
 #define VOS_BLK_SZ		(1UL << VOS_BLK_SHIFT) /* bytes */
 #define VOS_BLOB_HDR_BLKS	1	/* block */
 
+#define DTX_ARRAY_LEN		(1 << 15)	/* array length for local id */
```

Comment:
My main concern is whether 32K lid slots is enough for active DTX entries. Consider the case that: there are a lot of concurrent DTXs, some DTXs cannot become committable because of some daos target/replica is too busy or lost respond. Under such case, the lid slots for DTX on other replicas will be exhausted very quickly (may be within 1 ~ 2 seconds) before SWIM mechanism detected such issue and evicted the trouble daos target, and then the subsequent DTXs will get failure. That is not expected behavior.

Reply:
We can spare another 16 bits.   Obviously, there needs to be a cap though.  How many do you think we would need?

Reply:
It is not easy to say how many is enough. I would suggest to use dynamic LRU array or 2-level LRU array to avoid overflow in very short of time (may be 1 ~ 2 seconds).